README.md

```diff
@@ -21,13 +24,43 @@ you to produce a smaller build that works with the cosmwasm integration tests
 docker run --rm -v "$(pwd)":/code \
   --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
   --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
-  cosmwasm/rust-optimizer:0.9.0
+  cosmwasm/rust-optimizer:0.10.0
+
+ls -l ./artifacts/*.wasm
+cat ./artifacts/contracts.txt
 ```
 
+Demo this with `cosmwasm-examples` (going into eg. `erc20` subdir before running),
+with `cosmwasm-plus`, or with a sample app from `cosmwasm-template`.
+
 Note that we use one registry cache (to avoid excessive downloads), but the target cache is a different volume per
 contract that we compile. This means no interference between contracts, but very fast recompile times when making
 minor adjustments to a contract you had previously created an optimized build for.
 
+## Advanced usage
+
+*This is designed for cosmwasm samples. You cannot provide automatic verification for these*
+
+If you have a more complex build environment, you need to pass a few more
+arguments to define how to run the build process.
+
+[`cosmwasm`](https://github.com/CosmWasm/cosmwasm) has a root workspace
+and many contracts under `./contracts/*`, which are **excluded** in the
+top-level `Cargo.toml`. In this case, we compile each contract separately
+with it's own cache. However, since they may refer to packages via path
+(`../../packages/std`), we need to run the script in the repo root. In this
+case, we can use the optimize.sh command:
+
+```shell
+docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="devcontract_cache_burner",target=/code/contracts/burner/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer:0.10.0 ./contracts/burner
+
+ls -l ./artifacts/*.wasm
+cat ./artifacts/contracts.txt
+```
```

Comment:
We never documented this use case before and might even remove this feature from here if scripts get too complicated. I'd remove the entire block to keep it simple for users.